
From: Con Kolivas <kernel@kolivas.org>

The smt-nice handling is a little too aggressive by not estimating the per cpu
gain as high enough for pentium4 hyperthread.  This patch changes the per
sibling cpu gain from 15% to 25%.  The true per cpu gain is entirely dependant
on the workload but overall the 2 species of Pentium4 that support
hyperthreading have about 20-30% gain.

P.S: Anton - For the power processors that are now using this SMT nice 
infrastructure it would be worth setting this value separately at 40%.

Signed-off-by: Con Kolivas <kernel@kolivas.org>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/include/linux/sched.h |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

diff -puN include/linux/sched.h~sched-adjust-p4-per-cpu-gain include/linux/sched.h
--- 25/include/linux/sched.h~sched-adjust-p4-per-cpu-gain	2004-08-01 23:00:37.499832096 -0700
+++ 25-akpm/include/linux/sched.h	2004-08-01 23:00:37.503831488 -0700
@@ -621,7 +621,7 @@ struct sched_domain {
 	.imbalance_pct		= 110,			\
 	.cache_hot_time		= 0,			\
 	.cache_nice_tries	= 0,			\
-	.per_cpu_gain		= 15,			\
+	.per_cpu_gain		= 25,			\
 	.flags			= SD_BALANCE_NEWIDLE	\
 				| SD_BALANCE_EXEC	\
 				| SD_WAKE_AFFINE	\
_
